File: LeetCodeRecord/MethodFirst-Golang/154-FindMin.go
func findMin(nums []int) int {
	length := len(nums)
	
	left := 0
	right := length - 1

	for left < right {
		mid :=  left + (right - left) / 2
		
		if nums[mid] > nums[right] {
			left = mid + 1
		} else if nums[mid] < nums[right] {
			right = mid
		} else {
			index := left
			for i := left; i < right; i++ {
				if nums[index] > nums[i] {
					index = i
				}
			}

			return nums[index]
		}
	}

	return nums[left]
}

File: LeetCodeRecord/MethodFirst/154-FindMin.java
package MethodFirst;

class FindMin {
    public int findMin(int[] nums) {
        int len = nums.length;
        
        int left = 0;
        int right = len - 1;

        while (left < right) {
            int mid = left + ((right - left) >> 1);
            
            if (nums[mid] > nums[right]) {
                left = mid + 1;
            } else if (nums[mid] < nums[right]){
                right = mid;
            } else {
                int index = left;
                for (int i = left + 1; i < right; i++) {
                    if (nums[i] < nums[index]) {
                        index = i;
                    }
                }    
                
                return nums[index];
            }
        }

        return nums[left];
    }

    
}
